data/src/main/java/com/linkedin/data/collections/CheckedMap.java

```diff
@@ -149,14 +156,23 @@ public CheckedMap(int initialCapacity, MapChecker<K,V> checker)
   public CheckedMap(int initialCapacity, float loadFactor, MapChecker<K,V> checker)
   {
     _checker = checker;
-    _map = new HashMap<K,V>(initialCapacity, loadFactor);
+    _map = new HashMap<>(initialCapacity, loadFactor);
   }
 
   @Override
   public void clear()
   {
     checkMutability();
+    Set<K> keys = null;
+    if (_changeListeners != null)
+    {
+      keys = new HashSet<>(keySet());
+    }
     _map.clear();
+    if (keys != null)
+    {
+      notifyChangeListenersOnClear(keys);
```

Comment:
New to restli, why do we want to listen to data structure changes?

Reply:
So that we can clear our local cache of ivars, if the underlying map is updated directly instead of calling the wrapper record/union setters. The ivars have been added for performance reasons to avoid map lookups.